src/ad/azext_ad/tests/latest/test_ad_scenario.py

```diff
@@ -0,0 +1,144 @@
+# --------------------------------------------------------------------------
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License. See License.txt in the project root for
+# license information.
+#
+# Code generated by Microsoft (R) AutoRest Code Generator.
+# Changes may cause incorrect behavior and will be lost if the code is
+# regenerated.
+# --------------------------------------------------------------------------
+
+import os
+import time
+from azure.cli.testsdk import ScenarioTest
+from azure.cli.testsdk import ResourceGroupPreparer
+from .preparers import VirtualNetworkPreparer
+from .example_steps import step_ds_create
+from .example_steps import step_ds_list
+from .example_steps import step_ds_list2
+from .example_steps import step_ds_update
+from .example_steps import step_ds_show
+from .example_steps import step_ds_delete
+from .example_steps import step_ds_update_security_settings
+from .example_steps import step_ds_update_notification_settings
+from .. import (
+    try_manual,
+    raise_if,
+    calc_coverage
+)
+
+
+TEST_DIR = os.path.abspath(os.path.join(os.path.abspath(__file__), '..'))
+
+
+# Env setup_scenario
+@try_manual
+def setup_scenario(test, rg_2, rg):
+    test.kwargs.update({
+        'nsg': 'aadds-nsg-test',
+        'vn': 'vnet1',
+        'vn_2': 'vnet2'
+    })
+    test.cmd('az network nsg create --name {nsg} --resource-group {rg}')
+    test.cmd('az network nsg rule create -n AllowPSRemoting -g {rg} --nsg-name {nsg} --priority 301 '
+             '--access Allow --protocol Tcp --direction Inbound --destination-port-ranges 5986 '
+             '--source-address-prefixes AzureActiveDirectoryDomainServices')
+    test.cmd('az network nsg rule create -n AllowRD -g {rg} --nsg-name {nsg} --priority 201 '
+             '--access Allow --protocol Tcp --direction Inbound --destination-port-ranges 3389 '
+             '--source-address-prefixes CorpNetSaw')
+    test.cmd('az network vnet create -g {rg} -n {vn} --nsg {nsg} --subnet-name default '
+             '--address-prefixes 10.0.0.0/16')
+    test.cmd('az network vnet create -g {rg} -n {vn_2} --nsg {nsg} --subnet-name default '
+             '--address-prefixes 10.1.0.0/16')
```

Comment:
You could use VirtualNetworkPreparer to create vnet resource.

Reply:
The vnet need nsg rules configured but `VirtualNetworkPreparer` can't config nsg. So I just use `az network vnet create` directly since I have to use `az network nsg` anyway😀